src/lplib/cmap.ml

```diff
@@ -0,0 +1,172 @@
+(*Type of points and intervals. Used to determine if a cursor is in the range
+of a specific token, which is an interval of points (positions). *)
+(*The presence of t and compare make RangeType an OrderedType (in the sense of Map). *)
+module type RangeType = sig
+
+  type point
+  val make_point : int -> int -> point
+
+  type t
+  type interval = t
+  (*Ensures an interval is well defined (e.g with start < finish). *)
+  val make_interval : point -> point -> interval
+
+  val line : point -> int
+  val column : point -> int
+
+  val interval_start : interval -> point
+  val interval_end : interval -> point
+
+  type cmp = Before | In | After
+  val in_range : point -> interval -> cmp
+
+  (*Intervals need to be well defined (i.e returned by make_interval). *)
+  val compare : t -> t -> int
+
+  val point_to_string : point -> string
+  val interval_to_string : interval -> string
+
+end
+
+(*In this implementation, two intervals are considered 
+equal if one is included in the other.
+This is meant to locate a cursor (which is an interval 
+with start = finish) inside an interval
+
+In any other case, overlapping intervals
+can't be compared and will throw an error.
+
+An interval is considered "smaller" than another
+if all its points are smaller that the points
+of the other (i.e its finishing point is before
+the starting point of the other). *)
+module Range : RangeType = struct
+
+  type point = { l : int; c : int }
+  let make_point l c = { l = l; c = c}
+
+  type t = point * point
+  type interval = t
+  let make_interval x y = 
+    assert (x <= y);
+    (x, y)
+
+  let line pt = pt.l
+  let column pt = pt .c
+
+  let interval_start = fst
+  let interval_end = snd
+
+  type cmp = Before | In | After
+
```

Comment:
thanks to remove this kind of empty lines

Reply:
Isn't more readable?

Reply:
I'm not sure which lines I should remove here actually

Reply:
I'm not sure which lines I should remove here actually

Reply:
Done by automatic formatter.